magpie/api/management/resource/resource_views.py

```diff
@@ -59,6 +59,10 @@ def create_resource_view(request):
     resource_display_name = ar.get_multiformat_any(request, "resource_display_name", default=resource_name)
     resource_type = ar.get_value_multiformat_post_checked(request, "resource_type")
     parent_id = ar.get_value_multiformat_post_checked(request, "parent_id")
+    try:
+        parent_id = int(parent_id)
+    except ValueError:
+        pass
```

Comment:
This feels off. If the id is invalid, it should probably fail directly instead of silently passing.
Invalid id will simply be forwarded to `create_resource`.
Is this the possible source of the previous comment about `resource_id != None` for a service?

Reply:
I agree it feels off. But can you suggest another way to do it?
The problem is that the parent_id is a string according to the schema. Magpie calls this route itself in a couple places, and sometimes sends a string, sometimes an integer. `create_resource` in `magpie/api/management/resource/resource_utils.py` expects an integer, and returns the proper error message when the parent_id is invalid.

Reply:
Then we should update the schema to require an int.
Where is this route called that a string is passed? These should be fixed instead.

Reply:
Take a look at https://github.com/Ouranosinc/Magpie/pull/228/commits/44bc571e10b0954b9969f4b772d8773ee854a63c